config/example/node.yaml

```diff
@@ -36,12 +36,14 @@ node:
 
 grpc:
   - endpoint: s01.neofs.devenv:8080  # endpoint for gRPC server
+    conn_limit: 1  # connection limits; exceeding connection will not be declined, just blocked before active number decreases or client timeouts
```

Comment:
i dont think static value can be usable. How user will choose the exact value?

as a basis - quite possible, but then we still need to rely on the real load. For example, there is a difference between 20 clients constantly writing data and 10K ones sending a simple request (like healthcheck)

Reply:
can be discussed, i solved exactly the problem described in the issue: go routines number that lead to the OOM

> i dont think static value can be usable

i can say smth like that about almost every config value but we need to stick to smth

Reply:
> the trap is that we dont know what exact number cuz it's not const and depends on too many factors

but we need limits according to the issue. so you suggest a dynamic limit?

> static conn limit is not an option for this IMO

what is an option then?

> But lets focus on the introduced one

sure, i just meant that we should test it somehow (if we are ok with a static limit or course) and be able to change depending on the machine and possible load

Reply:
there are some proposed suggestions https://github.com/nspcc-dev/neofs-node/issues/1240#issuecomment-1084216928. That's all we have for now, and hard net conn limit isn't there

Reply:
> there are some proposed suggestions https://github.com/nspcc-dev/neofs-node/issues/1240#issuecomment-1084216928

i do not think returning some error is a good idea, i would prefer blocking instead and waiting for some timeout (on the client side) or succeeding with some delay

> hard net conn limit isn't there

that is here and in the issue's title

> hard net conn

once again, how do you suggest to make it non-hard?

Reply:
comment suggests at least 2 approaches

Reply:
Yeah, it's not easy to choose a proper value and the default might be unlimited, but this handle can be handy to have in case of some case-specific problems. Just like `MaxWebSocketClients` in https://github.com/nspcc-dev/neo-go/blob/master/docs/node-configuration.md#RPC-Configuration, just like `limit_conn` in nginx, just like https://pkg.go.dev/golang.org/x/net/netutil#LimitListener.

Reply:
ok will see how it'll gonna be used